File: Microservices/ListAPI/ListAPI/DTOs/ListDTO.cs


using ListAPI.Models;

namespace ListAPI.DTOs
{
    public class ListDTO : ErrorRequestBase, IRealTime
    {
        public int ListaId { get; set; }

        public int? UserId { get; set; }

        public bool ListaPublica { get; set; }

        public bool ListaStatus { get; set; }

        public string? ListaDescricao { get; set; }

        public string? ListaTitulo { get; set; }
        public bool IsOwner { get; set; }
        public Participant[] Participants { get; set; } = Array.Empty<Participant>();
        public int IdUserTarget { get; set; }
    }
}

File: src/front-app/services/authentication/lists/dtos/ListDTO.ts
export interface ListDTO {
    listaId: number;
    userId: number | null;
    listaPublica: boolean;
    listaStatus: boolean;
    listaDescricao: string | null;
    listaTitulo: string | null;
    idUserTarget: number;
    isOwner: boolean;
    participants: ListParticipant[];
}

export interface ListParticipant {
    idParticipant: number;
    listaId: number;
    email: string;
    userId: number;
}